packages/context-editor/src/utils/serialize.ts

```diff
@@ -0,0 +1,68 @@
+import { DOMParser, DOMSerializer, Node } from "prosemirror-model";
+
+import { baseSchema } from "../schemas";
+
+/**
+ * Serialize a ProseMirror node to HTML
+ *
+ * Requires an alternative document (eg JSDOM) if running in a non-browser environment
+ */
+export const prosemirrorToHTML = (
+	node: Node | { type: "doc"; content: any[] },
+	document?: Document
+): string => {
+	const doc = document ?? global.document;
+
+	if (!doc) {
+		throw new Error(
+			"No document specified. Please provide a document to the prosemirrorToHTML function. Environment: " +
+				(typeof window !== "undefined" ? "browser" : "node")
+		);
+	}
+
+	let toBeProcessedContent =
+		node instanceof Node ? node.content : Node.fromJSON(baseSchema, node).content;
+
+	const fragment = DOMSerializer.fromSchema(baseSchema).serializeFragment(toBeProcessedContent, {
+		document: doc,
+	});
+
+	const container = doc.createElement("div");
+	container.appendChild(fragment);
```

Comment:
no sure if there's a more elegant way to do this

Reply:
From about ten minutes of searching, no, it doesn't seem like there's a much better way.